### PR TITLE
envoy: Optimize list of allowed remote security IDs

### PR DESCRIPTION
**Pull request by jrajahalme**
We already omit explicit listing of allowed remote security IDs when they are
for a wildcard selector. Do the same if there is only one selector for
the port, as bpf l3/l4 will pass only allowed lDs through. Only if there
are multiple selectors, then we need to differentiate between remotes
based on the L7 rules, so we must explicitly list the allowed remotes
for non-wildcard selectors in that case.

1st commit contains only test refactoring to ease reviewing of the test changes in the 2nd commit.

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
Fixed over-eager optimization and rebased.

**Comment by jrajahalme** [comment]
test-me-please

File: pkg/envoy/server.go
Comment (christarazi) [inline pkg/envoy/server.go:67]: I assume we don't have any users using UDP for envoy and this won't break existing installations?

File: pkg/envoy/server_test.go
Comment (christarazi) [inline pkg/envoy/server_test.go:614]: Is this code still meant to be commented out?

**Review by christarazi — APPROVED** [review]
LGTM, minor questions. No need for second review.

**Review by aanm — APPROVED** [review]
Nothing else from my side besides what Chris point it out.

File: pkg/envoy/server.go
Comment (jrajahalme) [reply pkg/envoy/server.go:67]: Right. This will change in some future with QUIC/HTTP3.

File: pkg/envoy/server_test.go
Comment (jrajahalme) [reply pkg/envoy/server_test.go:614]: Yes. 1001 and 1002 are the only allowed remotes, but we (now) send it to Envoy as an L3 wildcard as the enforcement is already done in bpf and there are no L3-dependent L7 rules. The comment could help someone trying to work with this test case in future.

**Review by aditighag — APPROVED** [review]

**Pull request merged**
